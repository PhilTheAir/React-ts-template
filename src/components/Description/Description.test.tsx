import * as React from 'react';
import * as enzyme from 'enzyme';
import { Description } from '.';

it('renders the correct text when no enthusiasm level is given', () => {
  const description = enzyme.shallow(<Description countBy={2} />);
  expect(description).toMatchSnapshot();
});
