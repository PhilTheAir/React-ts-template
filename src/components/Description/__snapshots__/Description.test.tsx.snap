// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the correct text when no enthusiasm level is given 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Description
    countBy={2}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          My favorite number is 
          0
</p>,
        <button
          onClick={[Function]}
>
          Increase
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "My favorite number is ",
            0,
          ],
        },
        "ref": null,
        "rendered": Array [
          "My favorite number is ",
          0,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Increase",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Increase",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            My favorite number is 
            0
</p>,
          <button
            onClick={[Function]}
>
            Increase
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "My favorite number is ",
              0,
            ],
          },
          "ref": null,
          "rendered": Array [
            "My favorite number is ",
            0,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Increase",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Increase",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
